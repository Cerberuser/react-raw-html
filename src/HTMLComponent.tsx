import * as React from "react";

export interface HTMLComponentProps {
    rawHTML: string;
}

export class HTMLComponent extends React.Component<HTMLComponentProps> {

    render() {
        this.parseHTML();
        return <div dangerouslySetInnerHTML={{__html: this.props.rawHTML}}/>;
    }

    parseHTML() {
        const div = document.createElement("div");
        div.innerHTML = this.props.rawHTML;
        return div.childNodes;
    }

}
