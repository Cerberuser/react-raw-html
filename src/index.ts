export const HTMLComponent = 'To be created';
