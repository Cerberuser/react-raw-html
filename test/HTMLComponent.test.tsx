import * as React from "react";
import renderer from "react-test-renderer";
import { HTMLComponent } from "../src";

test("Raw HTML renders correctly", () => {
    const component = renderer.create(<HTMLComponent rawHTML="<span>Text</span>"/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
