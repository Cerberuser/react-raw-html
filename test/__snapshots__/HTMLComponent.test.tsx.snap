// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Raw HTML renders correctly 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<span>Text</span>",
    }
  }
/>
`;
